project/source_code/drivetrain/include/gearbox_simulator.h

```diff
@@ -1,19 +1,26 @@
 #ifndef GEARBOX_SIMULATOR_H
 #define GEARBOX_SIMULATOR_H
 #include <map>
+#include "user_input.h"
 
 class Gearbox {
     private:
-        std::pair<uint8_t, double> *gear_ratio;
+        double *gear_ratio;
         uint8_t gear_lever_position;
+        uint8_t gear_number;
+        uint8_t max_gear_number;
+
+        const uint8_t break_position_to_change_gear_lever = 10;
+        const uint16_t RPM_to_increase_gear_number = 5000;
+        const uint16_t RPM_to_decrease_gear_number = 2000;
```

Comment:
Values extracted from a table or just guessed? (What happens if we end up below 2000 rpm when increasing gear?)